src/rgw/rgw_op.cc

```diff
@@ -5332,3 +5333,350 @@ int RGWHandler::error_handler(int err_no, string *error_content) {
   // This is the do-nothing error handler
   return err_no;
 }
+
+int RGWCreateRole::verify_permission()
+{
+  if (s->auth_identity->is_anonymous()) {
+    return -EACCES;
+  }
+
+  if (!verify_user_permission(s, RGW_PERM_WRITE)) {
+    return -EACCES;
+  }
```

Comment:
most of `verify_permission()` is duplicated by all of the Role ops. could we factor this into a base class so each op only has to say whether it's a read or write? or maybe have a `RGWRoleRead` and `RGWRoleWrite` that the others inherit from?